source/module_io/cube_io.h

```diff
@@ -78,15 +124,15 @@ void write_cube(
      * @param nz the dimension of grids along z
      * @param data the interpolated results
      */
-    void trilinear_interpolate(std::ifstream& ifs,
+    extern void trilinear_interpolate(std::ifstream& ifs,
```

Comment:
as @mohanchen has commented and what I commented before, the use of extern seems not necessary.